homeassistant/components/nightscout/sensor.py

```diff
@@ -103,6 +104,11 @@ async def async_update(self):
             self._available = False
             _LOGGER.warning("Empty reply found when expecting JSON data")
 
+    def _get_bgl(self, value) -> float:
+        if self._unit_of_measurement == MMOL_L:
+            return round((value / MMOL_CONVERSION_FACTOR), 1)
+        return value
```

Comment:
We generally don't want these types of conversions or calculations in Home Assistant.

The reasoning is simple: Before we know it, we will have tons of integrations offering all kinds of conversions. If anything, such a problem needs to be solved globally and not on a case by case status.

So, at this point, we generally hold on to: Provide the uom the device natively provides.

So IMHO, we should not add this.

Reply:
Hey @frenck,

Unfortunately, the API of this integration only provides data in this specific unit of measure. 
It does have an endpoint indicating if the user would like to have the values displayed in a different UOM, but it does not handle the conversion itself. Changing the code on the integration would be much harder, as it is an(although not FDA approved) medical application, so the changes would be more drastic. 

I understand your point. Trying to support all different types of UOMs can get overwhelming and out of control, but all the applications that I know of that integrate with Nightscout, provide a way to convert the values since there are two different UOMs used worldwide for this type of data.

I've seen some requests on this repo, on the add-on repo, and also on the forums. That's why I thought it would be a good idea to implement the feature. Links below for reference:

- https://github.com/home-assistant/core/issues/45002
- https://github.com/marciogranzotto/addon-nightscout/issues/6 
- https://github.com/home-assistant/core/pull/41382#issuecomment-758675716
- https://community.home-assistant.io/t/nightscout-ability-to-choose-units/307571
- https://community.home-assistant.io/t/nightscout-plugin-show-mmol-l/316107

With that said, do you believe those would be legit reasons for the change?

Reply:
@frenck maybe is best to add this conversion on the `py_nightscout ` lib?